R/read-eval.R

```diff
@@ -68,7 +75,7 @@ parse_expr <- function(x) {
 parse_exprs <- function(x) {
   if (inherits(x, "connection")) {
     exprs <- parse(file = x)
```

Comment:
This logic doesn't seem right to me. I think you can ditch the `temporary` stuff, and just do:

```R
if (!isOpen(x)) {
  open(x)
  on.exit(close(x))
}
```

That's the convention that most base functions follow